librz/include/rz_analysis.h

```diff
@@ -739,23 +719,17 @@ typedef struct rz_analysis_var_access_t {
 	ut8 type; // RzAnalysisVarAccessType bits
 } RzAnalysisVarAccess;
 
-typedef struct rz_analysis_var_constraint_t {
-	_RzAnalysisCond cond;
-	ut64 val;
-} RzAnalysisVarConstraint;
-
 // generic for args and locals
 typedef struct rz_analysis_var_t {
 	RzAnalysisFunction *fcn;
 	char *name; // name of the variable
-	RzType *type; // type of the variable
 	RzAnalysisVarKind kind;
 	bool isarg;
 	int delta; /* delta offset inside stack frame */
 	char *regname; // name of the register
 	RzVector /*<RzAnalysisVarAccess>*/ accesses; // ordered by offset, touch this only through API or expect uaf
 	char *comment;
-	RzVector /*<RzAnalysisVarConstraint>*/ constraints;
+	RzConstrainedType *contype; // constrained type
```

Comment:
... such that here the `RzType *type` stays independent of its constraint. This will ease the access to a variable's type (which most of the time happens without caring for any constraints).